engine/src/main/scala/cromwell/engine/io/gcs/GcsBatchFlow.scala

```diff
@@ -54,7 +54,33 @@ class GcsBatchFlow(batchSize: Int, scheduler: Scheduler, onRetry: IoCommandConte
     }
   }
 
-  private val batchRequest: BatchRequest = {
+  /**
+    * Returns a new BatchRequest instance.
+    *
+    * The correlated errors that were seen were:
+    * - Via CROM-6708 timeouts were occurring, followed usually by...
+    * - Via CROM-6709 batches had grown beyond 3001+ requests.
+    *
+    * Examining BatchRequest.execute(), if any exception occurs during the method, either during the HTTP request or
+    * during one of the response handlers, the BatchRequest.execute() method simply throws the exception.
+    *
+    * When exceptions are NOT thrown the BatchRequest's internal queue is only partially or fully cleared:
+    * - partial : https://github.com/googleapis/google-api-java-client/blob/v1.31.1/google-api-client/src/main/java/com/google/api/client/googleapis/batch/BatchRequest.java#L275
+    * -   full  : https://github.com/googleapis/google-api-java-client/blob/v1.31.1/google-api-client/src/main/java/com/google/api/client/googleapis/batch/BatchRequest.java#L281
+    *
+    * The BatchRequest's internal queue is NOT cleared when exception are thrown inside BatchRequest.execute().
+    * Any subsequent enqueueing operations only add to the internal queue and do not replace it the internal queue.
+    * So eventually timeouts lead to large batches of 1000+ elements that cause further timeout exceptions too.
+    * Those failed batch elements were re-appended leading to batches of greater than 3001 requests.
```

Comment:
For the sanity of future maintainers it might be worth mentioning your finding that the "Received" number in the error message always seems to be 3000 even when the batch was known to have > 3000 elements:

```
413 Request Entity Too Large
POST https://storage.googleapis.com/batch/storage/v1
Inner request count exceeds the limit. Received: 3000, Limit: 3000
```